cmd/buildah/images.go

```diff
@@ -300,34 +306,63 @@ func outputImages(ctx context.Context, systemContext *types.SystemContext, store
 		}
 
 	outer:
-		for name, tags := range imageReposToMap(image.Names) {
-			for _, tag := range tags {
-				if !matchesReference(name+":"+tag, argName) {
+		for repo, tagsAndDigests := range imageReposToMap(image.Names) {
+			for _, tagOrDigest := range tagsAndDigests {
+				imageName := ""
+				imageDigest := ""
+				imageTag := ""
+				if repo != "" && tagOrDigest != "" {
+					if digest.Digest(tagOrDigest).Validate() == nil {
+						imageDigest = tagOrDigest
+						imageName = repo + "@" + imageDigest
+					} else {
+						imageTag = tagOrDigest
+						if len(image.Digest) > 0 {
+							imageDigest = image.Digest.String()
+						}
+						imageName = repo + ":" + imageTag
+					}
+				}
```

Comment:
All of the `imageName`/`imageDigest`/`imageTag`, and the full string format, was already present in the `reference.Named` object. The code has created that object, type-erased the `tagsAndDigest` field, and now laboriously reconstructs the values and the string format.

E.g. `imageName := namedRef.String()` is all it should take to create that value; the “has digest” check should be an interface conformance check, and so on.